src/WithNamedArguments.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM;
+
+interface WithNamedArguments
```

Comment:
Can we solve this issue without introducing empty marker interfaces?

Reply:
We can have `$em->createQuery('SELECT NEW NAMED CustomerDTO(...);`

But I find it less readable

Reply:
Do you prefer have a const value USE_NAMED_ARGS in dto instead of implents empty interface  ?

Reply:
The gods of PHP gave us attributes, so we don't need marker interfaces anymore. If you really really need to flag the DTO class in some way, please use an attribute for that.

That being said, I think I'd prefer a dedicated SQL syntax for named argument calls. That should make any marking of DTO classes obsolete.